sdk/core/core-lro/test/testClient.test.ts

```diff
@@ -57,6 +57,33 @@ describe("Long Running Operations - custom client", function() {
     assert.equal(result, "Done");
   });
 
+  if (isNode) {
+    it("won't throw UnhandledPromiseRejectionWarnings when poll called without pollUntilDone", async function() {
+      // NOTE: Don't set any responses so that poller.poll throws an error
+      const client = new TestClient(new TestTokenCredential("my-test-token"));
+      let foundUnhandled = false;
+      let checkerResolve: () => void;
+      const checkerPromise = new Promise((resolve) => {
+        checkerResolve = resolve;
+      });
+      const checker = () => {
+        foundUnhandled = true;
+        checkerResolve();
+      };
+
+      process.once("unhandledRejection", checker);
+      try {
+        await client.startLRO();
+        throw new Error("Test failure");
+      } catch (err) {
+        assert.notEqual(err.message, "Test failure", "client.startLRO did not throw an error.");
+        // delay(0) gives the event loop a chance emit the UnhandledPromiseRejectionWarning so we can catch it.
+        await Promise.race([checkerPromise, delay(0)]);
```

Comment:
couldn't we just do a sinon.fake() and check `.called` is false after doing an `await delay(0)`? Or even just set a boolean in the checker function.

Reply:
I can probably just do the `delay` now. I had checkerPromise when I was making sure it failed before fixing it, then needed the delay because it never completed with the fix in. Of course using just delay is the _obvious_ approach :)